File: core/client/game/src/commonGame/net/base/SceneBaseResponse.cs
using ShineEngine;

public abstract class SceneBaseResponse:BaseResponse
{
	/// <summary>
	/// me
	/// </summary>
	public Player me=GameC.player;

	public Scene scene;

	public SceneBaseResponse()
	{
		setNeedFullRead(ShineSetting.clientMessageUseFull);
		setNeedRelease();
	}

	protected override void preExecute()
	{
		if((scene=GameC.scene.getScene())==null)
		{
			Ctrl.warnLog("收到场景类消息时,场景不存在");
			return;
		}

		execute();
	}
}

File: core/server/project/commonSceneBase/src/main/java/com/home/commonSceneBase/net/base/SceneBaseResponse.java
package com.home.commonSceneBase.net.base;
import com.home.commonSceneBase.scene.base.BScene;
import com.home.shine.net.base.BaseResponse;

public abstract class SceneBaseResponse extends BaseResponse
{
	/** 主角id */
	protected long _playerID=-1;
	/** 场景 */
	public BScene scene;
	
	public SceneBaseResponse()
	{
		setNeedRelease();
	}
	
	public void setInfo(BScene scene,long playerID)
	{
		this.scene=scene;
		_playerID=playerID;
	}
	
	@Override
	protected void preExecute()
	{
		if(scene==null)
			return;
		
		doExecute();
	}
}
